pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -30,7 +30,9 @@ import (
 	"k8s.io/kubernetes/pkg/credentialprovider"
 )
 
+const chinaRegionPrefix = "cn-"
 const registryURLTemplate = "*.dkr.ecr.%s.amazonaws.com"
+const chinaRegistryURLTemplate = "*.dkr.ecr.%s.amazonaws.com.cn"
```

Comment:
amazonaws.cn is kind like aws.amazon.com for web visit and amazonaws.com.cn is the mirror site for amazonaws.com. It's really painful here to deal with the GFW related problems.